src/commands/schemas.ts

```diff
@@ -217,6 +222,107 @@ async function evolveSchemaSubjectCommand(subject: Subject) {
+ *
+ */
+async function deleteSchemaVersionCommand(schema: Schema) {
+  if (!(schema instanceof Schema)) {
+    logger.error("deleteSchemaVersionCommand called with invalid argument type", schema);
+    return;
+  }
+
+  const loader = ResourceLoader.getInstance(schema.connectionId);
+  const schemaGroup = await loader.getSchemasForSubject(schema.environmentId!, schema.subject);
+
+  // Ensure is still present in the registry / UI view gestured from wasn't stale.
+  const found = schemaGroup.find((s) => s.id === schema.id) !== undefined;
+  if (!found) {
+    vscode.window.showErrorMessage(`Schema not found in registry. View needs refresh?`);
+    logger.error(
+      `Schema version ${schema.version} not found in registry, cannot delete.`,
+      schemaGroup,
+    );
+    return;
+  }
+
+  // Determine if user wants to hard or soft delete.
+  const hardDelete = await hardDeletionQuickPick();
+  if (hardDelete === undefined) {
+    vscode.window.showErrorMessage("Schema deletion canceled.");
+    logger.info("User canceled schema version deletion.");
+    return;
+  }
+
+  // Ask if they are sure they want to delete the schema version.
+  const confirmation = await confirmSchemaVersionDeletion(hardDelete, schema, schemaGroup);
+
+  if (!confirmation) {
+    vscode.window.showErrorMessage("Schema deletion canceled.");
+    logger.info("User canceled schema version deletion.");
+    return;
+  }
+
+  let success = true;
+
+  // Drive the delete via the resource loader so will be cache consistent.
+  // Resource loader will also emit event to alert views to refresh if needed.
+  try {
+    const wasOnlyVersionForSubject = schemaGroup.length === 1;
+
+    // Delete the schema version. Will take care of clearing any internal
+    // caches.
+    await vscode.window.withProgress(
+      {
+        location: vscode.ProgressLocation.Notification,
+        title: `Deleting schema ...`,
+      },
+      async () => {
+        await loader.deleteSchemaVersion(schema, hardDelete, wasOnlyVersionForSubject);
+      },
+    );
+
+    let successMessage = `Version ${schema.version} of subject ${schema.subject} deleted.`;
+    if (wasOnlyVersionForSubject) {
+      successMessage += ` Subject ${schema.subject} deleted.`;
+    }
+    vscode.window.showInformationMessage(successMessage);
+
+    // Fire off event to update views if needed.
+    if (wasOnlyVersionForSubject) {
+      schemaSubjectChanged.fire({ subject: schema.subjectObject(), change: "deleted" });
+    } else {
+      schemaVersionsChanged.fire({ schema: schema, change: "deleted" });
+    }
+  } catch (e) {
+    success = false;
+    logError(e, "Error deleting schema version", undefined, true);
+    if (e instanceof Error) {
+      vscode.window.showErrorMessage(
+        `Error deleting schema version ${schema.version}: ${e.message}`,
+      );
+    } else {
+      vscode.window.showErrorMessage(`Error deleting schema version ${schema.version}: ${e}`);
```

Comment:
I'd also lean more toward using `showErrorNotificationWithButtons()` (with just the message strings) here since it will show the "Open Logs"/"File Issue" buttons by default